<?xml version="1.0"?>
<launch>
  <env name="ROSCONSOLE_CONFIG_FILE"
     value="$(find p3dx_2dnav)/config/custom_rosconsole.conf"/>
  <rosparam file="$(find p3dx_2dnav)/yamls/startAndGoalParams.yaml" command="load"/>

  <!-- Run the map server -->
  <arg name="map_yaml" default="$(find p3dx_2dnav)/maps/simulated/trial.yaml" />
  <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_yaml)" output="screen"/>

  <node pkg="tf2_ros" type="static_transform_publisher" name="link1_broadcaster" args="-8.8 -8.3 0 0 0 0 1 /map /new_map"/>

  <!-- Data Collection -->
  <node name="data" pkg="p3dx_2dnav" type="data.py"/>
  <!--<node name="obstacle" pkg="podi_move_base" type="obstacle"/>-->
  
  <!-- Start RVIZ -->
  <!-- <arg name="urdf" default="$(find amr_robots_description)/urdf/pioneer3dx.urdf"/>
  <param name="robot_description" textfile="$(arg urdf)" />
  <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" ></node>
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher" /> -->
  <include file="$(find p3dx_description)/launch/rviz.launch">
    <arg name="rviz_node_args" value="-d $(find p3dx_2dnav)/config/p3dx_2dnav.rviz"/>
  </include>
  <!--<node name="rviz" pkg="rviz" type="rviz" args="-d $(find podi_run_global_planners)/rviz/navigation.rviz" />-->
  
  <!--AMCL and GAZEBO-->
  <arg name="start_descriptor" default="test start 1" />
  <arg name="init_x" default="0" />
  <arg name="init_y" default="0" />
  <arg name="init_a" default="-1.57" />

  <!--AMCL-->
  <arg name="scan_topic" default="/p3dx/laser/scan"/>
  <!--<node name="amcl" pkg="amcl" type="amcl" args="">
    <remap from="scan" to="$(arg scan_topic)" />
    <param name="initial_pose_x" type="double" value="$(arg init_x )" />
    <param name="initial_pose_y" type="double" value="$(arg init_y )" />
    <param name="initial_pose_a" type="double" value="$(arg init_a )" />
    <param name="initial_cov_xx" type="double" value="0.01" />
    <param name="initial_cov_yy" type="double" value="0.01" />
    <param name="initial_cov_aa" type="double" value="0.01" />
  </node>-->
  

  <!--P3DX in GAZEBO-->
  <include file="$(find p3dx_gazebo)/launch/gazebo.launch">
    <arg name="gui" value="true"/>
    <arg name="init_x" value="8.2"/>
    <arg name="init_y" value="8.2"/>
    <arg name="init_a" value="-1.57"/>
    <arg name="contact" value="0" />
    <arg name="laser_noise" value="0.00"/>
  </include>

  <arg name="cmd_ns" default="p3dx" />
  <include file="$(find p3dx_2dnav)/launch/move_base.launch">
    <arg name="cmd_vel" value="/$(arg cmd_ns)/cmd_vel"/>
  </include>
  
  <!-- Start the TF Broadcasters -->
  <node pkg="podi_move_base" type="transform" name="transform" output="screen"/>
  <!--<node pkg="podi_move_base" type="lmsToBase" name="lmsToBase" output="screen"/>-->
  <!--<node pkg="p3dx_2dnav" type="transform.py" name="transformation" output="screen"/>-->
  
  <node pkg="podi_robot_human_coupled_planner" type="planner" respawn="false" name="coupled_planner" output="screen">
    <rosparam file="$(find podi_run_global_planners)/cfg/costmap_params.yaml" command="load" ns="costmap" />
    <rosparam file="$(find podi_run_global_planners)/cfg/global_planner_params.yaml" command="load"/>
  </node>

  <node pkg="podi_robot_only_planner" type="planner" respawn="false" name="robot_only_planner" output="screen">
    <rosparam file="$(find podi_run_global_planners)/cfg/costmap_params.yaml" command="load" ns="costmap" />
    <rosparam file="$(find podi_run_global_planners)/cfg/global_planner_params.yaml" command="load"/>
  </node>
  
  <node name="coupling_model_node" pkg="podi_navigation_helpers" type="coupling_model_node" output="screen">
    <rosparam file="$(find p3dx_2dnav)/yamls/coupling_model_and_robot_params.yaml" command="load"/>
  </node>

  <!--<node pkg="podi_run_global_planners" type="call_global_planners" name="podi_call_global_planners" output="screen"/>-->

</launch>
